File: 03_recursion/Haskell/02_greet.hs

greet2 name = putStrLn ("how are you, " ++ name ++ "?")

bye = putStrLn "ok bye!"

greet name = do
  putStrLn ("hello " ++ name)
  greet2 name
  putStrLn "getting ready to say bye..."
  bye

main = greet "adit"

File: 03_recursion/scala/02_greet.scala
def greet2(name: String): Unit = {
  println("how are you, " + name + "?")
}

def bye(): Unit = {
  println("ok bye!")
}

def greet(name: String): Unit = {
  println("hello, " + name + "!")
  greet2(name)
  println("getting ready to say bye...")
  bye()
}

greet("adit")
